core/trino-main/src/main/java/io/trino/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -588,7 +596,7 @@ private boolean isGroupingKey(Expression node)
                 return true;
             }
 
-            return groupingFields.contains(fieldId);
+            return getGroupingFields().contains(fieldId);
```

Comment:
How common are queries when aggregation analyzer is constructed but getGroupingFields is not invoked?

Reply:
I don't have answer to that question.
It happens for "my query", which is absurdly complicated and I cannot share it here. I also struggle to make up any query that will suffer the same problems on the same scale.
That is why my approach here is to help in "my case" which may or may not be common and, what's more important, not add any regression.